commit 3a4ac03ab880f2a5bdf4f7eba6ffe6b6ce5b12c0
Author: Kristof Provost <kristof@codepro.be>
Date:   Sun Aug 26 16:37:38 2018 +0200

    pf: Fix shutdown issue
    
    pfsync0 interfaces don't appear to notify us when they're destroyed
    (during vnet shutdown), which leads us to clean up (set if_pf_kif to
    NULL) in a freed struct ifnet.
    
    It's not quite clear if this is the correct fix, or if we need to work
    out why pf is not notified that pfsync0 was destroyed. This does fix the
    issues in the regression tests, unblocking further progress on pfsync
    performance.

diff --git sys/netpfil/pf/pf_if.c sys/netpfil/pf/pf_if.c
index cc9c2f80011..4e6bae4dd00 100644
--- sys/netpfil/pf/pf_if.c
+++ sys/netpfil/pf/pf_if.c
@@ -163,10 +163,6 @@ pfi_cleanup_vnet(void)
 	V_pfi_all = NULL;
 	while ((kif = RB_MIN(pfi_ifhead, &V_pfi_ifs))) {
 		RB_REMOVE(pfi_ifhead, &V_pfi_ifs, kif);
-		if (kif->pfik_group)
-			kif->pfik_group->ifg_pf_kif = NULL;
-		if (kif->pfik_ifp)
-			kif->pfik_ifp->if_pf_kif = NULL;
 		free(kif, PFI_MTYPE);
 	}
 
